types/framework/entities/activeEffect.d.ts

```diff
@@ -133,7 +133,10 @@ declare class ActiveEffect<P extends Actor | Item = Actor | Item> extends Embedd
    * @param options - Configuration options which modify the request.
    * @returns The updated ActiveEffect data.
    */
-  update(data: DeepPartial<ActiveEffect.Data>, options?: Entity.UpdateOptions): Promise<ActiveEffect.Data>;
+  update(
+    data: DeepPartial<ActiveEffect.Data> & Partial<Record<string, any>>,
```

Comment:
Wouldn't you want to make this a union type instead of an intersection type? From my understanding, you would provide either the regular object `DeepPartial<ActiveEffect.Data>` _or_ the dot notation. That would map to a union type of those two things because it allows you to pass both things as parameter. The way it is typed now, you can only pass objects that are both a `DeepPartial<ActiveEffect.Data>` _and_ a `Partial<Record<string, any>>`. This might also the reason for it not working with `unknown`.

Reply:
The problem with that approach is that `unknown | string` for example is just `unknown`

Reply:
What is the problem with that?

Reply:
Basically, if we make the typing `DeepPartial<D> | Partial<Record<string, unknown>>` it's the same thing as just using `Record<string, unknown>`. If we absolutely want to have the two completely separate, we'd need a conditional type of some sort.